### DISCUSSION TITLE
Build process & `node_modules` in production

### DISCUSSION THREAD

**Discussion by MaheshkumarSundaram** [neonexus/sails-react-bootstrap-webpack#84] [General]
Hi @neonexus ,

It may sound silly, but I wnated to make sure I understand the build process correctly. 

As we are building our front-end to the `public` folder, is it necessary to have `node_modules` in the production?

I believe the webapck will bundle the dependencies of front-end too. 

But to answer my own question, `node_modules` will be required for the dependencies of the backend (Sails) since the `npm run build` will only bundle front-end React code.

So, we should have the `node_modules` in the production for Sails.

Am I correct regarding the build process and the `node_modules` dependencies? If not, could you kindly brief me.

Thanks!

**Comment by neonexus** [comment]
So, there are 2 ways to go about `node_modules`... Leave it as is, or build your assets, and slim down `node_modules`.

You need the `devDependencies` installed in-order to build your `.tmp/public` folder assets. However, after you have built your front-end assets, you can now delete `node_modules` folder, and install just the production dependencies `npm i --production`, this will make your production server considerably smaller.

But yes, you need some form of `node_modules` to run anything.

**Reply by MaheshkumarSundaram** [reply]
Yes. Planned to do the same. Thanks!

**Comment by neonexus** [comment]
I'm re-opening this discussion, just so it's more easily visible.
